### ISSUE TITLE
[Bug] WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.

### ISSUE THREAD

**Issue by mobassir94** [coqui-ai/TTS#2089]
### Describe the bug

i tried to train vits from scratch for bangla by following this tutorial : https://github.com/coqui-ai/TTS/blob/dev/notebooks/Tutorial_2_train_your_first_TTS_model.ipynb
i tried to infer using checkpoint_194000.pth model(avg mle loss was around 15.xx ),the loss then never improved and it outputs audio file which sounds like male (even though i trained using female voice dataset) and also it outputs double length audio. for example the wav file i have attached below is of  16 seconds,when i use same sentence to infer using glowtts i get 0.8 seconds audio which is perfect but vits producing double length audio even with checkpoint_194000 weight file.
[out.zip](https://github.com/coqui-ai/TTS/files/9829007/out.zip)

is there anything wrong in my pipeline or there is something wrong with vits implementation for non-english dataset?
here is the config that i used for training vits
```

{'output_path': '/home/apsisdev/mobassir/data/tts/',
 'logger_uri': None,
 'run_name': 'vits_16_october',
 'project_name': None,
 'run_description': '🐸Coqui trainer run.',
 'print_step': 50,
 'plot_step': 100,
 'model_param_stats': False,
 'wandb_entity': None,
 'dashboard_logger': 'tensorboard',
 'log_model_step': None,
 'save_step': 500,
 'save_n_checkpoints': 5,
 'save_checkpoints': True,
 'save_all_best': False,
 'save_best_after': 10000,
 'target_loss': None,
 'print_eval': True,
 'test_delay_epochs': -1,
 'run_eval': True,
 'run_eval_steps': None,
 'distributed_backend': 'nccl',
 'distributed_url': 'tcp://localhost:54321',
 'mixed_precision': True,
 'epochs': 10000,
 'batch_size': 64,
 'eval_batch_size': 32,
 'grad_clip': [1000.0, 1000.0],
 'scheduler_after_epoch': True,
 'lr': 0.001,
 'optimizer': 'AdamW',
 'optimizer_params': {'betas': [0.8, 0.99],
  'eps': 1e-09,
  'weight_decay': 0.01},
 'lr_scheduler': '',
 'lr_scheduler_params': None,
 'use_grad_scaler': False,
 'cudnn_enable': True,
 'cudnn_deterministic': False,
 'cudnn_benchmark': True,
 'training_seed': 54321,
 'model': 'vits',
 'num_loader_workers': 4,
 'num_eval_loader_workers': 4,
 'use_noise_augment': False,
 'audio': {'fft_size': 1024,
  'sample_rate': 22050,
  'win_length': 1024,
  'hop_length': 256,
  'num_mels': 80,
  'mel_fmin': 0,
  'mel_fmax': None},
 'use_phonemes': False,
 'phonemizer': None,
 'phoneme_language': None,
 'compute_input_seq_cache': True,
 'text_cleaner': 'collapse_whitespace',
 'enable_eos_bos_chars': False,
 'test_sentences_file': '',
 'phoneme_cache_path': None,
 'characters': {'characters_class': 'TTS.tts.utils.text.characters.Graphemes',
  'vocab_dict': None,
  'pad': '<PAD>',
  'eos': '।',
  'bos': '<BOS>',
  'blank': '<BLNK>',
  'characters': 'ইগং়’ুঃন১ঝূও‘ঊোছপফৈ৮ষযৎঢঈকঠিজ০৬ীটডএঅঋধচে২৩ণউয়ঢ়খলভৗসহ্ড়দথবঔাঞশরৌম—ঐআৃঘঙ\u200cঁ৪৫ত',
  'punctuations': '?.,|!- ',
  'phonemes': None,
  'is_unique': True,
  'is_sorted': True},
 'add_blank': True,
 'batch_group_size': 0,
 'loss_masking': None,
 'min_audio_len': 1,
 'max_audio_len': inf,
 'min_text_len': 1,
 'max_text_len': inf,
 'compute_f0': False,
 'compute_linear_spec': True,
 'precompute_num_workers': 0,
 'start_by_longest': False,
 'datasets': [{'formatter': '',
   'dataset_name': 'ljspeech',
   'path': '/home/apsisdev/mobassir/data/tts/bn_tts/IndicTTS_Phase2_Bengali_fem_Speaker1_mono/mono/',
   'meta_file_train': '/home/apsisdev/mobassir/data/tts/bn_tts/IndicTTS_Phase2_Bengali_fem_Speaker1_mono/mono/metadata_female.txt',
   'ignored_speakers': None,
   'language': '',
   'meta_file_val': '',
   'meta_file_attn_mask': ''}],
 'test_sentences': [['পিপলস ইন্স্যুরেন্স অব চায়না ছেষট্টি বছর আগে ব্যবসা চালু করে।',
   'সোনার বাংলা আমি তোমায় ভালবাসি।']],
 'eval_split_max_size': None,
 'eval_split_size': 0.01,
 'use_speaker_weighted_sampler': False,
 'speaker_weighted_sampler_alpha': 1.0,
 'use_language_weighted_sampler': False,
 'language_weighted_sampler_alpha': 1.0,
 'use_length_weighted_sampler': False,
 'length_weighted_sampler_alpha': 1.0,
 'model_args': {'num_chars': 86,
  'out_channels': 513,
  'spec_segment_size': 32,
  'hidden_channels': 192,
  'hidden_channels_ffn_text_encoder': 768,
  'num_heads_text_encoder': 2,
  'num_layers_text_encoder': 6,
  'kernel_size_text_encoder': 3,
  'dropout_p_text_encoder': 0.1,
  'dropout_p_duration_predictor': 0.5,
  'kernel_size_posterior_encoder': 5,
  'dilation_rate_posterior_encoder': 1,
  'num_layers_posterior_encoder': 16,
  'kernel_size_flow': 5,
  'dilation_rate_flow': 1,
  'num_layers_flow': 4,
  'resblock_type_decoder': '1',
  'resblock_kernel_sizes_decoder': [3, 7, 11],
  'resblock_dilation_sizes_decoder': [[1, 3, 5], [1, 3, 5], [1, 3, 5]],
  'upsample_rates_decoder': [8, 8, 2, 2],
  'upsample_initial_channel_decoder': 512,
  'upsample_kernel_sizes_decoder': [16, 16, 4, 4],
  'periods_multi_period_discriminator': [2, 3, 5, 7, 11],
  'use_sdp': True,
  'noise_scale': 1.0,
  'inference_noise_scale': 0.667,
  'length_scale': 1.0,
  'noise_scale_dp': 1.0,
  'inference_noise_scale_dp': 1.0,
  'max_inference_len': None,
  'init_discriminator': True,
  'use_spectral_norm_disriminator': False,
  'use_speaker_embedding': False,
  'num_speakers': 0,
  'speakers_file': None,
  'd_vector_file': None,
  'speaker_embedding_channels': 256,
  'use_d_vector_file': False,
  'd_vector_dim': 0,
  'detach_dp_input': True,
  'use_language_embedding': False,
  'embedded_language_dim': 4,
  'num_languages': 0,
  'language_ids_file': None,
  'use_speaker_encoder_as_loss': False,
  'speaker_encoder_config_path': '',
  'speaker_encoder_model_path': '',
  'condition_dp_on_speaker': True,
  'freeze_encoder': False,
  'freeze_DP': False,
  'freeze_PE': False,
  'freeze_flow_decoder': False,
  'freeze_waveform_decoder': False,
  'encoder_sample_rate': None,
  'interpolate_z': True,
  'reinit_DP': False,
  'reinit_text_encoder': False},
 'lr_gen': 0.0002,
 'lr_disc': 0.0002,
 'lr_scheduler_gen': 'ExponentialLR',
 'lr_scheduler_gen_params': {'gamma': 0.999875, 'last_epoch': -1},
 'lr_scheduler_disc': 'ExponentialLR',
 'lr_scheduler_disc_params': {'gamma': 0.999875, 'last_epoch': -1},
 'kl_loss_alpha': 1.0,
 'disc_loss_alpha': 1.0,
 'gen_loss_alpha': 1.0,
 'feat_loss_alpha': 1.0,
 'mel_loss_alpha': 45.0,
 'dur_loss_alpha': 1.0,
 'speaker_encoder_loss_alpha': 1.0,
 'return_wav': True,
 'use_weighted_sampler': False,
 'weighted_sampler_attrs': None,
 'weighted_sampler_multipliers': None,
 'r': 1,
 'num_speakers': 0,
 'use_speaker_embedding': False,
 'speakers_file': None,
 'speaker_embedding_channels': 256,
 'language_ids_file': None,
 'use_language_embedding': False,
 'use_d_vector_file': False,
 'd_vector_file': None,
 'd_vector_dim': 0,
 'restore_path': '/home/apsisdev/mobassir/data/tts/vits_16_october/checkpoint_176000.pth',
 'github_branch': 'inside_docker'}
```

is checkpoint_194000  too early to decide? but after few more epochs i start getting warning like this : 

WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.

followed by NaN loss as shown here -> https://github.com/coqui-ai/TTS/issues/2070
```

--> STEP: 29/104 -- GLOBAL_STEP: 217050
| > loss_disc: nan (nan)
| > loss_disc_real_0: nan (nan)
| > loss_disc_real_1: nan (nan)
| > loss_disc_real_2: nan (nan)
| > loss_disc_real_3: nan (nan)
| > loss_disc_real_4: nan (nan)
| > loss_disc_real_5: nan (nan)
| > loss_0: nan (nan)
| > grad_norm_0: 0.00000 (0.00000)
| > loss_gen: nan (nan)
| > loss_kl: nan (nan)
| > loss_feat: nan (nan)
| > loss_mel: 15.34929 (16.04639)
| > loss_duration: nan (nan)
| > amp_scaler: 0.00000 (0.00000)
| > loss_1: nan (nan)
| > grad_norm_1: 0.00000 (0.00000)
| > current_lr_0: 0.00015
| > current_lr_1: 0.00015
| > step_time: 1.05040 (0.99310)
| > loader_time: 0.01950 (0.02812)

```

@erogol

original sample rate of my dataset is 48000, in VitsAudioConfig() there is no resample = True parameter but i used :

```
audio_config = VitsAudioConfig(
    sample_rate=22050, win_length=1024, hop_length=256, num_mels=80, mel_fmin=0, mel_fmax=None
)

```
is this a probablem?
i tried with bigger batch size like 112 in A6000 gpu and still i get double length audio like the one attached above.

### To Reproduce

i used training script like this -> https://github.com/coqui-ai/TTS/blob/dev/recipes/ljspeech/vits_tts/train_vits.py for bangla

### Expected behavior

shouldn't get double length audio(compared to original), also shouldn't get NaN during training

### Logs

```shell
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.
WARNING:tensorboardX.x2num:NaN or Inf found in input tensor.

followed by NaN loss as shown here -> https://github.com/coqui-ai/TTS/issues/2070


--> STEP: 29/104 -- GLOBAL_STEP: 217050
| > loss_disc: nan (nan)
| > loss_disc_real_0: nan (nan)
| > loss_disc_real_1: nan (nan)
| > loss_disc_real_2: nan (nan)
| > loss_disc_real_3: nan (nan)
| > loss_disc_real_4: nan (nan)
| > loss_disc_real_5: nan (nan)
| > loss_0: nan (nan)
| > grad_norm_0: 0.00000 (0.00000)
| > loss_gen: nan (nan)
| > loss_kl: nan (nan)
| > loss_feat: nan (nan)
| > loss_mel: 15.34929 (16.04639)
| > loss_duration: nan (nan)
| > amp_scaler: 0.00000 (0.00000)
| > loss_1: nan (nan)
| > grad_norm_1: 0.00000 (0.00000)
| > current_lr_0: 0.00015
| > current_lr_1: 0.00015
| > step_time: 1.05040 (0.99310)
| > loader_time: 0.01950 (0.02812)

```
```


### Environment

```
TTS 0.8.0
gpu : rtx 3090
pytorch : 1.10
```


### Additional context

_No response_

**Comment by mobassir94** [comment]
sorry,
i had bug in my pipeline.
my dataset had sample rate 48000  but in config i was using 22050 instead without resampling.
i just downsampled my dataset down to 22050 and also used text_cleaner=None, instead of "collapse_whitespace" and now vits training working fine,so far so good,didn't observe any NaN yet,also checked inference with best_model.pth and it is doing as good job as expected,still the model is getting trained,hopefully i won't face any NaN or other issue anymore,if yes then i might will reopen this issue,thanks
